plugin/src/functionalTest/kotlin/io/qalipsis/gradle/cloud/PublishScenarioTaskFunctionalTest.kt

```diff
@@ -0,0 +1,241 @@
+package io.qalipsis.gradle.cloud
+
+import io.kotest.matchers.string.shouldContain
+import io.qalipsis.gradle.MockserverBasedTest
+import io.qalipsis.gradle.SSETestServer
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import org.testcontainers.junit.jupiter.Testcontainers
+import java.nio.file.Path
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.temporal.ChronoUnit
+import java.util.stream.Stream
+import kotlin.io.path.writeText
+
+@Testcontainers
```

Comment:
Those annotations could land on MockserverBasedTest to avoid repeating them on each and every subclass.